src/angular-mighty-datepicker.coffee

```diff
@@ -95,17 +96,37 @@ angular.module("mightyDatepicker").directive "mightyDatepicker", ["$compile", ($
       switch $scope.options.mode
         when "multiple"
           return _indexOfMoment($scope.model, day, 'day') > -1
+        when "range"
+          if $scope.model
+            return day.isSame($scope.model.start, 'day')
         else
           return $scope.model && day.isSame($scope.model, 'day')
 
-    _isInRange = (day) ->
-      if $scope.options.rangeMode
-        if $scope.options.rangeMode == "from"
-          return moment.range($scope.model, $scope.before).contains(day) || day.isSame($scope.before, 'day')
+    _isSelectedTo = (day) ->
+      switch $scope.options.mode
+        when "range"
+          if $scope.model
+            return day.isSame($scope.model.end, 'day')
         else
-          return moment.range($scope.after, $scope.model).contains(day) || day.isSame($scope.after, 'day')
-      else
-        return false
+          return false
+
+    _isInRange = (day) ->
+      switch $scope.options.mode
+        when "multiple"
+          if $scope.options.rangeMode
+            if $scope.options.rangeMode == "from"
+              return moment.range($scope.model, $scope.before).contains(day) ||
+                  day.isSame($scope.before, 'day')
+            else
+              return moment.range($scope.after, $scope.model).contains(day) ||
+                  day.isSame($scope.after, 'day')
+          else
+            return false
+        when "range"
+          if $scope.model.start
```

Comment:
@spaceribs you assume that model holds a valid range here.
It's possible to initialise the picker with empty date/range and this line causes an error